src/modules/addNewChain/components/NewAddress.tsx

```diff
@@ -0,0 +1,148 @@
+import { FC, useState } from 'react';
+
+import { useSelector } from 'react-redux';
+import { MdError } from 'react-icons/md';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+
+import useToast from 'hooks/useToast';
+import { useInitiateNewChain } from 'queries';
+
+import { chainLabels } from '../AddNewChain.constants';
+import { LOGO_ALIAS_CHAIN } from 'common';
+
+// Components
+import { Box, Button, Select, TextInput } from 'blocks';
+
+import { isValidAddress } from 'helpers/ValidationHelper';
+import { convertAddressToAddrCaip } from 'helpers/CaipHelper';
+
+import { NewChainAddressValue } from '../AddNewChain.types';
+import { UserStoreType } from 'types';
+
+export type NewAddressProps = {
+  setAddress: React.Dispatch<React.SetStateAction<string>>;
+  handleNextStep: () => void;
+};
+
+const NewAddress: FC<NewAddressProps> = ({ setAddress, handleNextStep }) => {
+  const [selectedChainValue, setSelectedChainValue] = useState<string>('80002');
+  const { userPushSDKInstance } = useSelector((state: UserStoreType) => {
+    return state.user;
+  });
+
+  const toast = useToast();
+
+  const { mutate: initiateNewChain, isPending, isError } = useInitiateNewChain();
+
+  //optimise this
+  const selectOptions = Object.keys(LOGO_ALIAS_CHAIN).map((key) => {
+    const chainId = parseInt(key, 10);
+    const Component = LOGO_ALIAS_CHAIN[chainId];
+    return {
+      value: key,
+      label: chainLabels[chainId],
+      icon: <Component />,
+    };
+  });
+
+  console.debug(selectOptions, 'chain');
+
+  const validationSchema = yup.object().shape({
+    alias: yup
+      .string()
+      .required('Address is required')
+      .test('is-valid-address', 'Invalid wallet address', isValidAddress),
+  });
+  const formik = useFormik<NewChainAddressValue>({
+    initialValues: {
+      alias: '',
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values) => {
+      setAddress(convertAddressToAddrCaip(values.alias, parseInt(selectedChainValue)));
+      handleInitiate(values.alias);
+    },
+  });
```

Comment:
all the form related stuff like types, initial values and validation schema needs to go to AddNewChain.form.ts